lib/discourse_theme/cli.rb

```diff
@@ -167,6 +183,94 @@ def run(args)
         else
           UI.info "Manage: #{client.root}/admin/customize/themes/#{theme_id}"
         end
+      elsif command == "rspec"
+        index = dir.index("/spec")
+        spec_path = "/spec"
+
+        if index
+          spec_path = dir[index..-1]
+          dir = dir[0..index - 1]
+        end
+
+        raise DiscourseTheme::ThemeError.new "'#{dir} does not exist" unless Dir.exist?(dir)
+
+        # Checks if the container is running
+        container_name = DISCOURSE_TEST_DOCKER_CONTAINER_NAME
+        is_running = false
+        container_exists = false
+
+        if !(
+             output =
+               execute(command: "docker ps -a --format '{{json .}}' | grep #{container_name}")
+           ).empty?
+          container_exists = true
+          is_running = JSON.parse(output)["State"] == "running"
+        end
+
+        basename = Pathname.new(dir).basename.to_s
+        verbose = !!args.delete("--verbose")
+        headless = !!args.delete("--headless")
+
+        if !is_running || args.delete("--rebuild")
+          if container_exists
+            execute(command: "docker stop #{container_name}")
+            execute(command: "docker rm -f #{container_name}")
+          end
+
+          execute(
+            command: <<~CMD.squeeze(" "),
+              docker run -d \
+                -p 31337:31337 \
+                --add-host host.docker.internal:host-gateway \
+                --entrypoint=/sbin/boot \
+                --name=#{container_name} \
+                -v #{dir}:/tmp/#{basename} \
+                discourse/discourse_test:release
+            CMD
+            message: "Creating discourse/discourse_test:release Docker container...",
+            stream: verbose,
+          )
+
+          execute(
+            command:
+              "docker exec -u discourse:discourse #{container_name} ruby script/docker_test.rb --no-tests --checkout-ref origin/tests-passed",
+            message: "Checking out latest Discourse source code...",
+            stream: verbose,
+          )
+
+          execute(
+            command:
+              "docker exec -u discourse:discourse #{container_name} bundle exec rake docker:test:setup",
+            message: "Setting up Discourse test environment...",
```

Comment:
This takes a very long time on my machine (probably because it's virtualising an x86 container on an M1 mac). Would be nice if there was some kind of feedback on what it's doing? (I guess stuff like bundle install, db create/migrate, etc?)

Reply:
Ah I see there is a verbose flag... so maybe that's good enough for now. Still would be a 'nice to have' if it could print the main milestones in non-verbose mode.